llvm/tools/sycl-post-link/CompileTimePropertiesPass.cpp

```diff
@@ -142,9 +152,124 @@ PreservedAnalyses CompileTimePropertiesPass::run(Module &M,
     }
   }
 
+  // Check pointer annotations.
+  SmallVector<IntrinsicInst *, 4> RemovableAnnots;
+  for (Function &F : M)
+    for (BasicBlock &BB : F)
+      for (Instruction &Inst : BB)
+        if (auto *IntrInst = dyn_cast<IntrinsicInst>(&Inst))
+          if (IntrInst->getIntrinsicID() == Intrinsic::ptr_annotation &&
+              transformSYCLPropertiesAnnotation(M, IntrInst, RemovableAnnots))
+            CompileTimePropertiesMet = true;
+
+  // Remove irrelevant "sycl-properties" annotations after the transformations.
+  for (IntrinsicInst *IntrInst : RemovableAnnots) {
+    assert(IntrInst->getNumUses() == 0);
+    IntrInst->eraseFromParent();
+  }
+
   // The pass just adds some metadata to the module, it should not ruin
   // any analysis, but we need return PreservedAnalyses::none() to inform
   // the caller that at least one compile-time property was met.
   return CompileTimePropertiesMet ? PreservedAnalyses::none()
                                   : PreservedAnalyses::all();
 }
+
+bool CompileTimePropertiesPass::transformSYCLPropertiesAnnotation(
```

Comment:
Could you add a comment what this function returns. As I get it, the function returns `true` if it changes the module (re-writes at least one intrinsic). Is it correct?

Reply:
That's right. I have added a comment,. Is this what you had in mind?

Reply:
Thank you. Yes, it is.